ichnaea/locate/provider.py

```diff
@@ -531,3 +545,63 @@ class GeoIPPositionProvider(BaseGeoIPProvider):
 
 class GeoIPCountryProvider(BaseGeoIPProvider):
     location_type = Country
+
+
+class FallbackProvider(Provider):
+    log_name = 'fallback'
+    location_type = Position
+    source = DataSource.Fallback
+
+    @property
+    def fallback_url(self):
+        return read_config().get_map('ichnaea')['fallback_url']
+
+    def should_locate(self, data, location):
+        empty_location = not location.found()
+        weak_location = location.source >= DataSource.GeoIP
+        cell_found = data.get('cell', None)
+        wifi_found = len(data.get('wifi', [])) > 1
+        return (
+            self.api_key.allow_fallback and
+            (empty_location or weak_location) and
+            (cell_found or wifi_found)
+        )
+
+    def locate(self, data):
+        location = self.location_type(query_data=False)
+
+        cell_queries = [{
+            'cellId': cell.get('cid', None),
+            'locationAreaCode': cell.get('lac', None),
+            'mobileCountryCode': cell.get('mcc', None),
+            'mobileNetworkCode': cell.get('mnc', None),
+        } for cell in data.get('cell', [])]
+        wifi_queries = [{
+            'macAddress': re.sub(
+                '(..(?!$))', '\g<0>:', wifi.get('key', None)),
+        } for wifi in data.get('wifi', [])]
+
+        query_data = {
+            'cellTowers': cell_queries,
+            'wifiAccessPoints': wifi_queries,
+        }
+
+        try:
+            response = requests.post(
+                self.fallback_url, json=query_data)
+        except requests.exceptions.RequestException:
+            response = None
+
+        if response:
+            try:
+                location_data = json.loads(response.content)
+            except ValueError:
```

Comment:
Why not just use `response.json` - via http://docs.python-requests.org/en/latest/user/quickstart/#json-response-content